### DISCUSSION TITLE
Looking to see if possible to build against maya-2024.2.0 and mayaUSD-0.27.0 on Linux

### DISCUSSION THREAD

**Discussion by TxRxFx** [Autodesk/maya-hydra#130] [General]
I'm trying to see if it's possible to build this plugin against `maya-2024.2.0` and `mayaUSD-0.27.0` on Linux. I believe I've taken it up to the point where it's looking for newer versions than  `usd-0.22.11`.

Is there a branch or tag I should use with those specific versions of Autodesk tools?

**Comment by lanierd-adsk** [comment]
Hi,
Sorry for the late reply. 
Though we don't officially support compiling MayaHydra with maya 2024, we are working an helper document on how to compile with maya 2024 and a custom version of OpenUSD. 
This helper document is being done compiling with Windows but it should be very similar for Linux.

But note that OpenUSD, MayaUSD and MayaHydra should be compiled with the same version of Python and MayaUSD and MayaHydra should always use the same OpenUSD version to be able to load usd files under a Hydra render delegate in maya's viewport.

For Maya 2024 and using a custom version of OpenUSD, if they don't use the same Python version as Maya 2024, OpenUSD will have to be recompiled with the suitable version of Python. Maya 2024  used Python 3.10.8.
I will keep you informed when this helper document is published.
Regards,
David

**Comment by lanierd-adsk** [comment]
You may have a look at this document :
https://github.com/Autodesk/maya-hydra/blob/dev/doc/rebuildingWithCustomOpenUSDAndPreviousMayaVersion.md 
Regards,
David
